src/Provider/Exception/IdentityProviderException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace League\OAuth2\Client\Provider\Exception;
+
+class IdentityProviderException extends \Exception
+{
+    protected $result;
+
+    public function __construct($message, $code, $result)
+    {
+        $this->result = $result;
```

Comment:
What exactly is result going to be? An array, an object? Is it just a raw value we're allowing the provider to pass around so the user can access it on the exception?